caas/kubernetes/provider/k8s.go

```diff
@@ -1619,93 +1657,141 @@ func (k *kubernetesClient) Units(appName string) ([]caas.Unit, error) {
-				logger.Debugf(fmt.Sprintf("PersistentVolumeClaim for %v is pending", vol.PersistentVolumeClaim.ClaimName))
+			if fsInfo == nil {
 				continue
 			}
-			pv, err := pVolumes.Get(pvc.Spec.VolumeName, v1.GetOptions{})
-			if k8serrors.IsNotFound(err) {
-				// Ignore volumes which don't exist (yet).
-				continue
-			}
-			if err != nil {
-				return nil, errors.Annotate(err, "unable to get persistent volume")
-			}
-
-			storageName := pvc.Labels[labelStorage]
-			if storageName == "" {
+			if fsInfo.StorageName == "" {
 				if valid := legacyJujuPVNameRegexp.MatchString(volMount.Name); valid {
-					storageName = legacyJujuPVNameRegexp.ReplaceAllString(volMount.Name, "$storageName")
+					fsInfo.StorageName = legacyJujuPVNameRegexp.ReplaceAllString(volMount.Name, "$storageName")
 				} else if valid := jujuPVNameRegexp.MatchString(volMount.Name); valid {
-					storageName = jujuPVNameRegexp.ReplaceAllString(volMount.Name, "$storageName")
+					fsInfo.StorageName = jujuPVNameRegexp.ReplaceAllString(volMount.Name, "$storageName")
 				}
 			}
-			statusMessage := ""
-			since = now
-			if len(pvc.Status.Conditions) > 0 {
-				statusMessage = pvc.Status.Conditions[0].Message
-				since = pvc.Status.Conditions[0].LastProbeTime.Time
-			}
-			if statusMessage == "" {
-				// If there are any events for this pvc we can use the
-				// most recent to set the status.
-				events := k.CoreV1().Events(k.namespace)
-				eventList, err := events.List(v1.ListOptions{
-					IncludeUninitialized: true,
-					FieldSelector:        fields.OneTermEqualSelector("involvedObject.name", pvc.Name).String(),
-				})
-				if err != nil {
-					return nil, errors.Annotate(err, "unable to get events for PVC")
-				}
-				// Take the most recent event.
-				if count := len(eventList.Items); count > 0 {
-					statusMessage = eventList.Items[count-1].Message
-				}
-			}
-
-			unitInfo.FilesystemInfo = append(unitInfo.FilesystemInfo, caas.FilesystemInfo{
-				StorageName:  storageName,
-				Size:         uint64(vol.PersistentVolumeClaim.Size()),
-				FilesystemId: string(pvc.UID),
-				MountPoint:   volMount.MountPath,
-				ReadOnly:     volMount.ReadOnly,
-				Status: status.StatusInfo{
-					Status:  k.jujuFilesystemStatus(pvc.Status.Phase),
-					Message: statusMessage,
-					Since:   &since,
-				},
-				Volume: caas.VolumeInfo{
-					VolumeId:   pv.Name,
-					Size:       uint64(pv.Size()),
-					Persistent: pv.Spec.PersistentVolumeReclaimPolicy == core.PersistentVolumeReclaimRetain,
-					Status: status.StatusInfo{
-						Status:  k.jujuVolumeStatus(pv.Status.Phase),
-						Message: pv.Status.Message,
-						Since:   &since,
-					},
-				},
-			})
+			unitInfo.FilesystemInfo = append(unitInfo.FilesystemInfo, *fsInfo)
 		}
 		units = append(units, unitInfo)
 	}
 	return units, nil
 }
 
+func (k *kubernetesClient) volumeInfoForEmptyDir(vol core.Volume, volMount core.VolumeMount, now time.Time) (*caas.FilesystemInfo, error) {
```

Comment:
is it better to name it `since` here?

Reply:
We tend to use "now" in other places when setting timestamps on things.
When the timestamp is reported in status etc, it is labelled "Since".